nexus_common/src/action_client_bt_node_test.cpp

```diff
@@ -16,7 +16,7 @@
  */
 
 #define CATCH_CONFIG_MAIN
-#include <catch2/catch.hpp>
+#include <rmf_utils/catch.hpp>
```

Comment:
What is the rationale for this change?

Also, please upgrade `rmf_utils` to `<build_depend>` https://github.com/osrf/nexus/blob/728fd0d366f9bc65da9c318d64a0c5e8c385cd89/nexus_common/package.xml#L25

Reply:
I haven't actually looked into it but I did now! Reason, compilation fails without:

```
--- stderr: nexus_common                                                                                                                                                                                
[...]/src/nexus/nexus_common/src/action_client_bt_node_test.cpp:19:10: fatal error: catch2/catch.hpp: No such file or directory
   19 | #include <catch2/catch.hpp>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
```

Explanation:
The [file vendored](https://github.com/open-rmf/rmf_utils/blob/main/rmf_utils/include/rmf_utils/catch.hpp) by `rmf_utils` is indeed in the `rmf_utils/catch.hpp` path.
However, there is a `catch2` [package in rosdep](https://github.com/ros/rosdistro/blob/1f6d3a742ce750d3c0bb13797d8e1fc870a2d825/rosdep/base.yaml#L470-L485). We don't directly depend on it but if anyone in the dependency tree was it would be installed.
That package, in its [Ubuntu 22.04 version](https://packages.ubuntu.com/jammy/amd64/catch2/filelist) _provides_ a `catch2/catch.hpp` file, however in the [Ubuntu 24.04 version](https://packages.ubuntu.com/noble/amd64/catch2/filelist) it doesn't.
Hence in my opinion it was really working by pure chance and we should indeed include the vendored one

Reply:
WRT upgrading to `build_depend`, it seems there are two categories:

* Packages that use the `impl_ptr`, for which there is a `<depend>rmf_utils</depend>`
* Packages that only use rmf_utils for uncrustify, for which the find_package call is under a `if (BUILD_TESTING)` statement, those have it as a `<test_depend>`.

With this in mind do you still want to change them to a `build_depend`? They are not really needed unless we build tests

Reply:
Thanks for explaining the reason for the header change.

According to [REP-0140](https://www.ros.org/reps/rep-0140.html#build-depend-multiple), I would say this should be a `build_depend` since `rmw_utils` is need at build time to supply the header needed for compilation.

Although under the `test_depend` description it's stated "When building with testing enabled, the <test_depend> packages are available for configuring and building the tests as well as running them." which suits are requirement.

I'm leaning towards `build_depend` here but let me know what you think based on `REP-0140`.

Reply:
I went through the packages and made it a `build_depend` of the ones that include the `catch.hpp` header https://github.com/osrf/nexus/pull/35/commits/ca6ef25d2b36052dcc847bc5ee7e7a98865d613c